### PR TITLE
Merge feature/carrier into master

### PR DESCRIPTION
**Pull request by JamesGallicchio**
- Adds carrier functionality
- Adds remote control for carrier

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -43,7 +45,7 @@ int rightMotorSpeed = 0;
 void moveWheels(int wheelPower)
 {
 	leftMotorSpeed = wheelPower;
-	rightMotorSpeed = -wheelPower-1;
+	rightMotorSpeed = -wheelPower-1;	//why is there a -1?
```
Comment (JamesGallicchio) [inline main.c:48]: Was this resolved?

Should it be here? If yes, remove comment or explain why it's here. If no, remove the -1.

File: main.c [hunk 2]
```diff
@@ -173,24 +175,24 @@ void updateClaw()
 	//5D lowers arm
 void updateArm()
 {
-	int power = 0;
+	//int power = 0;
```
Comment (JamesGallicchio) [inline main.c]: Remove this if it isn't being used

File: main.c [hunk 3]
```diff
@@ -173,24 +175,24 @@ void updateClaw()
...
 
 	if(vexRT[Btn5D] == 1) //if left lower Z button is pressed
 	{
-		power += 1;
-		motor[armLeft] = -7-power;
-		motor[armRight] = 6+power;
+		//power += 1;
```
Comment (JamesGallicchio) [inline main.c]: Remove if isn't being used

File: main.c [hunk 4]
```diff
@@ -173,24 +175,24 @@ void updateClaw()
...
+		motor[armLeft] = -8;	//-7-power;
+		motor[armRight] = 7;	//6+power;
 	}
-	else if((SensorValue[pot] <= 0)||((vexRT[Btn5U] == 0) && (vexRT[Btn5D] == 0)))
+	else if(((SensorValue[pot]+2000) <= 0)||((vexRT[Btn5U] == 0) && (vexRT[Btn5D] == 0)))
```
Comment (JamesGallicchio) [inline main.c]: Could you explain each part of this in a comment on the line above this? the logic is a bit hard to follow- something like `// pot < -2000 (carrier is retracted) OR both up and down buttons are pressed` would suffice (not sure if I'm interpreting it correctly)

File: main.c [hunk 5]
```diff
@@ -173,24 +175,24 @@ void updateClaw()
...
 	{
 		motor[armLeft] = -30;
 		motor[armRight] = 30;
 	}
   else if(vexRT[Btn5U] == 1) //if left upper Z button is pressed
 	{
-		power += 60;
-		motor[armLeft] = -50-power;
-		motor[armRight] = 50+power;
+		//power += 60;
```
Comment (JamesGallicchio) [inline main.c]: Clean up these comments here. Maybe add a comment saying what the motor powers are doing i.e. `// Lift arm` or whatevs

File: main.c [hunk 6]
```diff
@@ -199,22 +201,68 @@ void updateArm()
 void raiseCarrier()
 {
 	//move carrier motors at speed so it should difinitely be able to lift the mobile goal
-	motor[carrier] = 100;
+	//for(int coefficient = 1; coefficient <= 10; coefficient++)
```
Comment (JamesGallicchio) [inline main.c]: Remove this if it's not being used

File: main.c [hunk 7]
```diff
@@ -199,22 +201,68 @@ void updateArm()
...
+	//}
+	for(int base = 1; base <= 10; base++)
+	{
+		motor[carrierLeft] = pow(base,2);
+		motor[carrierRight] = -pow(base,2);
+		sleep(186);
+	}
 	//wait 1 second, needs to be tested and probably changed but this is the test value
-	sleep(1000);
+	//sleep(1000);
```
Comment (JamesGallicchio) [inline main.c]: Remove this and the comment above if it's not being used

File: main.c [hunk 8]
```diff
@@ -199,22 +201,68 @@ void updateArm()
...
 	//stop carrier motors
-	motor[carrier] = 0;
+	motor[carrierLeft] = 0;
+	motor[carrierRight] = 0;
 }
 
+//lowers the mobile goal, needs to be heavily tested
```
Comment (JamesGallicchio) [inline main.c]: This comment is here twice

File: main.c [hunk 9]
```diff
@@ -199,22 +201,68 @@ void updateArm()
...
 //lowers the mobile goal, needs to be heavily tested
 void lowerCarrier()
 {
+	//for(int coefficient = 1; coefficient <= 4; coefficient++)
```
Comment (JamesGallicchio) [inline main.c]: Same as above- remove if it's not being used

File: main.c [hunk 10]
```diff
@@ -199,22 +201,68 @@ void updateArm()
...
+		motor[carrierRight] = (pow(base,2));
+		sleep(105);
+	}
 	//move carrier motors at the same descending speed the arm has, 6
-	motor[carrier] = -50;
+	//motor[carrierRight] = -50;
+	//motor[carrierLeft] = -50;
 	//wait 1 second, needs to be tested and probably changed but this is the test value
-	sleep(1000);
+	//sleep(1000);
```
Comment (JamesGallicchio) [inline main.c]: Same as before, remove if it's not being used

File: main.c [hunk 11]
```diff
@@ -199,22 +201,68 @@ void updateArm()
...
+	//8R lowers mobile goal
+void updateMobileGoal()
+{
+	if(up){
+		if(vexRT[Btn8R] == 1){
+			lowerCarrier();
+			up = false;
+		}
+	}
+	else{
```
Comment (JamesGallicchio) [inline main.c:246]: Clean this up to `else if { ... }` (unnecessary brackets and indentation)

**Review by JamesGallicchio — COMMENTED** [review]
Overall well done changes! Lots of comments in your code, which I like. Most of the requested changes are just cleaning up stuff.

(To fix any requested changes, just edit the `feature/carrier` branch and push your changes)

File: main.c (hunk 1 above)
Comment (COBrien19) [reply main.c:48]: It was not resolved so I just kept it. Ill fix it if I have time.

File: main.c (hunk 2 above)
Comment (COBrien19) [reply main.c]: It is no longer being used, but it doesn't matter because it isn't carrier-related

File: main.c (hunk 3 above)
Comment (COBrien19) [reply main.c]: it isn't carrier related idc what happens to it

File: main.c (hunk 4 above)
Comment (COBrien19) [reply main.c]: I could, but it's not carrier related so I'll do it in the arm branch

File: main.c (hunk 5 above)
Comment (COBrien19) [reply main.c]: ok i will in arm code

File: main.c (hunk 6 above)
Comment (COBrien19) [reply main.c]: will do

File: main.c (hunk 7 above)
Comment (COBrien19) [reply main.c]: Ok

File: main.c (hunk 8 above)
Comment (COBrien19) [reply main.c]: I'll remove it

File: main.c (hunk 9 above)
Comment (COBrien19) [reply main.c]: On it

File: main.c (hunk 10 above)
Comment (COBrien19) [reply main.c]: k

File: main.c (hunk 11 above)
Comment (COBrien19) [reply main.c:246]: I'll do it after the competition
